server/src/document/document.service.ts

```diff
@@ -74,6 +89,40 @@ export class DocumentService {
     }
   }
 
+  async findArtifactSharepointDocuments(artifactName, id: string) {
+    try {
+      const strippedArtifactName = stripDcpName(artifactName);
+      const folderIdentifier = constructFolderIdentifier(strippedArtifactName, id);
+
+      const { value: documents } = await this.crmService.getSharepointFolderFiles(`dcp_artifacts/${folderIdentifier}`);
+
+      if (documents) {
+        return documents.map(document => ({
+          name: document['Name'],
+          timeCreated: document['TimeCreated'],
+          serverRelativeUrl: document['ServerRelativeUrl'],
+        }));
+      }
+
+      return [];
+    } catch (e) {
+      // Relay errors from crmService 
+      if (e instanceof HttpException) {
+        throw e;
+      } else {
+        throw new HttpException({
+          code: 'SHAREPOINT_FOLDER_ERROR',
+          title: 'Bad Sharepoint folder lookup',
+          detail: `An error occured while constructing and looking up folder for artifact. Perhaps the artifact name or id is wrong.`,
+          meta: {
+            artifactName: artifactName,
+            artifactsId: id,
+          }
+        }, HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Again, where is the inclusion of `e.message`? You use a template literal on 116, but no string interpolation.

Again, the biggest issue I've had with these errors, including in applicant portal, are how they conceal the actual error message. "An error occured" — okay, but what's the error? It's right there in the catch block argument.

I would think more critically about your own experience using this error handling system. Has it been frustrating? What could make it better?

Reply:
Ah, good catch.. should have included that. I updated the exception detail to include `e`.
I decided to log and include the entire, `JSON.stringify()"-ed `e` because it sometimes contains a `code` property which might be valuable to debugging. 

I also updated the other functions to make it consistent across the service 

Yeah. I think it needs a concentrated effort to improve it,  so am going to dedicate it to a separate task.